### ISSUE TITLE
Unifying linearInterpolation and blend (CPU)

### ISSUE THREAD

**Issue by davidpiuva** [magic-lang/ooc-kean#1333]
I was about to refactor linearInterpolation and blend so that vectors and colors both follow the GLSL standard called `mix`. The problem is that the float library now has banned the use of static functions for some reason so now I have to break convention to unify our math library. Now colors linterpolate with the first value as owner while vectors use the ratio as owner. The mix function is a merging of two values combined with a ratio setting and letting any of them be the owner makes no sense to me. Now we have four different standards for linear interpolation in our codebase and only three of them can be changed.

**Colors**
Here we use the first of two colors as the owner and the other is just an argument of lesser importance. This is **not consistent** with the Float interpolation because they have different arguments as owners.
```ooc
blend: func (factor: Float, other: This) -> This { This new((this y * (1 - factor) + (other y * factor)) as Byte) }
```

**Floats** (According to the new convention against static functions)
Here we use the ratio as the owner which feels completely backwards. We **can't even apply** this for vectors since the owner always has the same Float type.
```ooc
linearInterpolation: func (a: This, b: This) -> This { (this * (b - a)) + a }
```

**Vectors**
Here we just follow the math standard. This could be shorter by not belonging to a cover.
```ooc
linearInterpolation: static func (a, b: This, ratio: Float) -> This {
	This new(ratio mix(a x, b x), ratio mix(a y, b y))
}
```

**GLSL**
Here we just follow the math standard.
```
float mix(float x, float y, float a)
```

**Comment by davidpiuva** [comment]
Since overloading is a way to tie a function to a type instead of using name-spaces, we can even make the math functions global instead of static to save space when writing.

**Comment by marcusnaslund** [comment]
Fixed by #1761
